src/Game/GamePage.js

```diff
@@ -3,27 +3,35 @@ import "./Game.css";
 import AttackButton from "./AttackButton";
 import { supabase } from "../supabase_client";
 import AttackTargetSelect from "./AttackTargetSelect";
+import PlayerLocations from "./PlayerLocations";
 
 const GamePage = (props) => {
   const { roomCode, playerName } = props;
   const [totalPlayerData, setTotalPlayerData] = useState([]);
   const [attackTarget, setAttackTarget] = useState("Select a target");
   const [currentTurnPlayerIndex, setCurrentTurnPlayerIndex] = useState(0);
+  const [currentPlayerLocationID, setCurrentPlayerLocationID] = useState(0);
+  const [hasPlayerMovedLocations, setHasPlayerMovedLocations] = useState(false);
```

Comment:
we use this boolean to imply a lot of things right? like if they can move and attack is disabled or if they can attack after moving. it might be easier to introduce a phase state variable that can be "attack", or "location". I think that might make the code more clear in places you use this variable